src/Interpreters/AsynchronousMetrics.cpp

```diff
@@ -1343,43 +1343,19 @@ void AsynchronousMetrics::update(std::chrono::system_clock::time_point update_ti
         new_values["TotalRowsOfMergeTreeTables"] = total_number_of_rows;
         new_values["TotalPartsOfMergeTreeTables"] = total_number_of_parts;
 
-        auto get_metric_name = [](const String & name) -> const char *
-        {
-            static std::map<String, const char *> metric_map =
-            {
-                {"tcp_port", "TCPThreads"},
-                {"tcp_port_secure", "TCPSecureThreads"},
-                {"http_port", "HTTPThreads"},
-                {"https_port", "HTTPSecureThreads"},
-                {"interserver_http_port", "InterserverThreads"},
-                {"interserver_https_port", "InterserverSecureThreads"},
-                {"mysql_port", "MySQLThreads"},
-                {"postgresql_port", "PostgreSQLThreads"},
-                {"grpc_port", "GRPCThreads"},
-                {"prometheus.port", "PrometheusThreads"}
-            };
-            auto it = metric_map.find(name);
-            if (it == metric_map.end())
-                return nullptr;
-            else
-                return it->second;
-        };
-
         if (servers_to_start_before_tables)
         {
             for (const auto & server : *servers_to_start_before_tables)
             {
-                if (const auto * name = get_metric_name(server.getPortName()))
-                    new_values[name] = server.currentThreads();
+                new_values["Interface_" + server.getInterfaceName()] = server.currentThreads();
```

Comment:
That requires compatibility notes (old metrics / and monitoring around will stop working).

Reply:
Probably. However, that seems to be a fairly new addition.